messaging/src/main/java/org/axonframework/eventhandling/pooled/PooledStreamingEventProcessor.java

```diff
@@ -399,6 +386,32 @@ private void statusUpdater(int segmentId, UnaryOperator<TrackerStatus> segmentUp
         processingStatus.computeIfPresent(segmentId, (s, ts) -> segmentUpdater.apply(ts));
     }
 
+    private class TokenStoreInitializer implements Runnable {
+
+        @Override
+        public void run() {
+            executeUntilTrue(this::initializeTokenStore, 100L);
+        }
+
```

Comment:
I am wondering why you've kept the runnable in the PSEP and didn't move the entire method to the `Coordinator`.
Is that because of the `initialSegmentCount` and `initialToken` variables, perhaps?